### PR TITLE
Update eos-dev dependencies

### PR DESCRIPTION
**Pull request by dsd**
Add an explanatory comment in eos-dev-depends to explain what types of
packages should be listed there.

Remove the following packages which I assume do not meet these guidelines:
bison, bonnie++, elinks, flex, fluxbox, germinate, glade, gparted,
gperf, language-pack-*, lftp, libjson-glib-doc, locate, ltrace, moreutils,
nethogs, nmap, putty, sl, tig

gnome-api-docs is not present in the distro.

devhelp is available as a flatpak.

openssh-server and p7zip-full are included in the main distro

Switch pyflakes to python3 version

Philip C suggests that eos-node-modules-dev is probably not used
by anyone any more, since you can do node development on unconverted
systems.

Add some packages that I think should be here:
manpages, obs-build, pbuilder

Create and document a new eos-dev-suggests file, to list packages
that do not meet the eos-dev-depends guidelines, but should not be
removed from the distro as they are still valuable developer tools for
limited/occasional use.
These packages will then not be considered for deletion from our distro
if they are not needed by anything else.

### CODE REVIEW

**Comment by dsd** [comment]
@dbnicholson please review this and merge if you agree with the approach.
After merge my plan is to send an endless-dev email highlighting this change in case I made bad assumptions, and giving people a chance to state their interest in the ~400 unused packages that I've identified can be removed from OBS, before I go further in the distro purge.

**Comment by dbnicholson** [comment]
Wow, thank you for doing this. eos-dev was such an eyesore. This was basically one person's idea of "here's a bunch of stuff developers might want".

We should also consider making the eosdev ostree and images again after these changes, but that's a separate topic.

File: eos-metapackage [hunk 1]
```diff
@@ -86,6 +86,7 @@ sub subst_depends {
 foreach my $package (@{$dh{DOPACKAGES}}) {
 	subst_depends($package, "depends", "eos:Depends");
 	subst_depends($package, "recommends", "eos:Recommends");
+	subst_depends($package, "suggests", "eos:Suggests");
```
Comment (dbnicholson) [inline eos-metapackage]: Can you put this change in a separate commit? Probably doesn't matter, but it's independent and should stay even if we have to revert the rest of this for some reason.

**Comment by dsd** [comment]
updated now
